### ISSUE TITLE
v4.14 doesn't compatible with Browserify

### ISSUE THREAD

**Issue by chatii2412** [lodash/lodash#2527]
Lodash doesn't work with Browserify after update to v4.14 because there aren't `global` variable unlike Node.js.

https://github.com/lodash/lodash/blob/master/dist/lodash.js#L364

``` js
/** Detect free variable `global` from Node.js. */
var freeGlobal = typeof global == 'object' && global && global.Object === Object && global;
```

**Comment by jdalton** [comment]
Are you saying it throws an error?
Can you provide a code snippet or execution?

I don't have anything actionable here.

**Comment by chatii2412** [comment]
Sorry for my insufficient report, I was trying to find exact problem.

The problem is that Lodash v4.14.0 can't read `global` variable if Browserify's options are set as `{insertGlobals: true, detectGlobals: false}`.
I think it should be fixed or documentated somewhere.

**Comment by chatii2412** [comment]
Sorry for my insufficient report, I was trying to find exact problem.

The problem is that Lodash v4.14.0 can't read `global` variable when Browserify's options are set as `{insertGlobals: true, detectGlobals: false}`.

I think it should be fixed or documentated somewhere.

```
$ browserify index.js --insertGlobals true --detectGlobals false -o foo.js

node:
```
$ node foo.js
> 4.14.0
```

in browser:
```
undefined
```

**Comment by jdalton** [comment]
Neat!

This is because `freeExports` requires `freeGlobal` in v4.14.0 where it didn't in v4.13.1.
I'm OK with this for now. I'll address it if it starts tripping up lots of folks.

**Comment by jdalton** [comment]
Looks like  a Browserify bug too as `--insertGlobals true` should insert the `global` object.

From its [readme](https://github.com/substack/node-browserify/blob/master/readme.markdown#browserifyfiles--opts):
> When `opts.insertGlobals` is true, always insert `process`, `global`,
`__filename`, and `__dirname` without analyzing the AST for faster builds but
larger output bundles. Default false.

**Comment by HaroldStruwig** [comment]
This issue is giving me a hard time in NW.js. I am using browserify with the --node option. In v4.14.0 I get an empty object whenever I require lodash. In v4.13.1 it works as expected.

**Comment by jdalton** [comment]
@HaroldStruwig 
The browserify docs state:
>   --bare
>
    Alias for both --no-builtins, --no-commondir, and sets --insert-global-vars
    to just "__filename,__dirname". This is handy if you want to run bundles in
    node.
>
  --no-browser-field, --no-bf
>
    Turn off package.json browser field resolution. This is also handy if you
    need to run a bundle in node.
>
  --node
>
    Alias for --bare and --no-browser-field.

I'm guessing you're wanting it to export like node `module.exports` path but NW.js lacks a `global` object? If that's the case then the `--node` flag is not appropriate for NW.js.

**Comment by justiceo** [comment]
If you need some extra motivation to address this, it would be worth of note that as many applications that use "^4.13.1" instead of "4.13.1" (or anything below) would break.

**Comment by polpo** [comment]
This also just tripped us up majorly. LIke @HaroldStruwig we are using NW.js and browserifying with the --node option. Lodash winds up just being an empty object. Can this be reopened?

**Comment by marudor** [comment]
Jest ist sill broken with 4.14.1

**Comment by jdalton** [comment]
I haven't published 4.14.1?

**Comment by jdalton** [comment]
Try replacing the `node_modules/lodash/lodash.js` with [this file](https://github.com/lodash/lodash/blob/master/lodash.js).
